src/client-functions/selector-builder/add-api.js

```diff
@@ -223,7 +237,17 @@ function addFilterMethods (obj, getSelector, SelectorBuilder) {
     };
 }
 
-function createHierachicalSelector (getSelector, SelectorBuilder, selectorFn, filter, additionalDependencies) {
+function addSnapshotExtensionMethods (obj, getSelector, SelectorBuilder) {
+    obj.extendSnapshot = extensions => {
```

Comment:
We are getting rid of snapshot term in the first place. Maybe we'll rename it to `addCustomDomProperty`?